catalog/transformer/catalog-transformer-geojson-queryresponse/src/main/java/ddf/catalog/transformer/queryresponse/geojson/GeoJsonQueryResponseTransformer.java

```diff
@@ -63,17 +66,33 @@
     }
   }
 
-  public static JSONObject convertToJSON(Result result) throws CatalogTransformerException {
+  private MetacardTransformer metacardTransformer;
```

Comment:
Good point. I'll implement your default metacard approach.

Reply:
If you are worried about about state and such why not change this to be a final property and be passed as a constructor argument?

We should be using blueprint to construct beans. Also since this is a service, other classes should not be creating new instances to begin with - they should be just using the service.

Reply:
True, making the `MetacardTransformer` `final` would make it stateless, but it still doesn't eliminate the possibility of passing a `GeoJsonQueryResponseTransformer` to code that expects transformations to behave in a different way. Adding a new parameter will ensure that only new code will be affected while maintaining the old method's behavior and that the change in behavior will be completely explicit right at the `transform` method call in new code. It also avoids the inconvenience of having to construct a new `GeoJsonQueryResponseTransformer` to call `transform` if you need to pass in a specific `MetacardTransformer`.

That being said, though, you probably know more about how this class interacts with the blueprint system. If it needs a `MetacardTransformer` to be a constructor parameter in order to be accessible to blueprints, then so be it.

Reply:
@rzwiefel that's also a valid solution. To be honest, I overlooked setting the transformer as a constructor arg 😅 . But if we did go with the default approach, the intent was to still pull in the default transformer as a service in the blueprint. It wouldn't be instantiated here.

@Variadicism I'm not sure I understand your concern about eliminating "the possiblity of passing a `GeoJsonQueryResponseTransformer` to code that expects transformations to behave a certain way". By definition, the metacard transformer would dictate the behavior. If we pass in the `GeoJsonMetacardTransformer` here then the behavior would be the same as before and no one would be the wiser. If another `GeoJsonQueryResponseTransformer` is used somewhere else with a different metacard transformer, then only the behavior of that specific service would change.

Reply:
I going to go with the `MetacardTransformer` as a constructor approach. All in favor?